applications/ShapeOptimizationApplication/python_scripts/algorithms/algorithm_gradient_projection.py

```diff
@@ -270,9 +270,13 @@ def __getActiveConstraints(self):
     def __isConstraintActive(self, constraint):
         identifier = constraint["identifier"].GetString()
         constraint_value = self.communicator.getStandardizedValue(identifier)
-        if constraint["type"].GetString() == "=":
-            return True
-        elif constraint_value >= 0:
+        if constraint["type"].GetString() == "=" or constraint_value >= 0:
+            gradient_norm = self.optimization_utilities.ComputeMaxNormOfNodalVariable(
+                self.design_surface, self.constraint_gradient_variables[identifier]["mapped_gradient"]
+            )
+            if gradient_norm <= 1e-16:
```

Comment:
```suggestion
            if math.isclose(gradient_norm, 0.0):
```

probably even better, I recently learned abt this

Reply:
since the reference here is 0, it will do the same - comparing absolute value.

Reply:
```suggestion
            if math.isclose(gradient_norm, 0.0, abs_tol = 1e-16):
```
as a compromise =)

Reply:
There seems to be much love for the isclose function =). I am convinced 🤣